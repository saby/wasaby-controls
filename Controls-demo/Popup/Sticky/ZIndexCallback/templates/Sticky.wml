<Controls.popupTemplate:Sticky headingCaption="Заголовок"
                               headingFontColorStyle="warning"
                               headingFontSize="xl">
    <ws:bodyContentTemplate>
        <div class="controlsDemo-PopupTemplate__list-item controlsDemo_fixedWidth200 controlsDemo__wrapper">
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Animi, blanditiis consequatur dolorum eos error
            fuga hic inventore iure labore minima nihil nobis porro provident quaerat quibusdam repudiandae similique?
            Excepturi, voluptatum!
        </div>
    </ws:bodyContentTemplate>
</Controls.popupTemplate:Sticky>
