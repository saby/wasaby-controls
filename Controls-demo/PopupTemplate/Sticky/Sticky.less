.controlsDemo-PopupTemplate__footer {
   font-size: 13px;
   padding: 0 12px;
   color: #aaa;
   border-top: 1px dashed #aaa;
}

.controlsDemo-PopupTemplate__list-item {
   height: 32px;
   display: flex;
   align-items: center;
}
