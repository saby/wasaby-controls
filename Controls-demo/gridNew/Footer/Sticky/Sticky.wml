<div class="controlsDemo__wrapper controlDemo__grid-header-sticky">
    <Controls.scroll:Container attr:class="controlsDemo__height400">
        <Controls.grid:View
                keyProperty="key"
                source="{{_viewSource}}"
                stickyFooter="{{true}}"
                columns="{{_columns}}">
            <ws:footerTemplate>
                <ws:partial template="Controls/grid:FooterTemplate">
                    <div>Контент подвала таблицы</div>
                </ws:partial>
            </ws:footerTemplate>
        </Controls.grid:View>
    </Controls.scroll:Container>
</div>
