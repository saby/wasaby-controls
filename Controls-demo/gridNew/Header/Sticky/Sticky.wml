<div class="controlsDemo__wrapper controlDemo__grid-header-sticky">
    <Controls.scroll:Container attr:class="controlsDemo__height400 controlsDemo__width800px">
        <Controls.grid:View
                backgroundStyle="default"
                storeId="HeaderSticky"
                header="{{_header}}"
                columns="{{_columns}}"/>
    </Controls.scroll:Container>
</div>
