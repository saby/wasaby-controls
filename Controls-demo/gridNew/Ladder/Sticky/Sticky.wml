<div class="controlsDemo__wrapper">
    <Controls.scroll:Container attr:class="controlsDemo__height400" shadowMode="js">
        <Controls.listDeprecate:Mover name="listMover"/>
        <Controls.grid:View
				name="grid"
                keyProperty="key"
                source="{{_viewSource}}"
                columns="{{_columns}}"
                itemsDragNDrop="{{true}}"
                on:customdragStart="_dragStart()"
                on:customdragEnd="_dragEnd()"
                multiSelectVisibility="visible"
                bind:selectedKeys="_selectedKeys"
                header="{{_header}}"
                resultsPosition="top"
                ladderProperties="{{_ladderProperties}}">
			<ws:footerTemplate>
				<ws:partial template="Controls/grid:FooterTemplate">
					<Controls.list:AddButton caption="Add record" on:click="_beginAdd()" data-qa="controlsDemo-GridLadder__addButton"/>
				</ws:partial>
			</ws:footerTemplate>
		</Controls.grid:View>
    </Controls.scroll:Container>
</div>
