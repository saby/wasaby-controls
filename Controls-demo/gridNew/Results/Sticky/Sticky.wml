<div class="controlsDemo__wrapper controlsDemo_fixedWidth800">
    <Controls.scroll:Container attr:class="controlsDemo__height400">
        <Controls.grid:View
                backgroundStyle="default"
                storeId="ResultsSticky"
                header="{{_header}}"
                resultsPosition="top"
                columns="{{_columns}}"/>
    </Controls.scroll:Container>
</div>
