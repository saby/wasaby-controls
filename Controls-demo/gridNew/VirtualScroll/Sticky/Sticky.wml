<div class="controlsDemo__wrapper controlsDemo_fixedWidth500">
    <span class="controls-text-label controlsDemo__mb1">pageSize 50, остальные опции не заданы, показ с первой страницы</span>
    <Controls-demo.list_new.VirtualScroll.NavigationSwitcher.Index storeId="VirtualScrollSticky" totalCount="{{500}}" pageSize="{{50}}">
        <ws:content>
            <Controls.scroll:Container attr:style="height: 300px;">
                <Controls.grid:View
                        storeId="VirtualScrollSticky1"
                        columns="{{_columns}}"
                        header="{{_header}}"
                        groupProperty="group">
                    <ws:virtualScrollConfig pageSize="{{30}}"/>
                </Controls.grid:View>
            </Controls.scroll:Container>
        </ws:content>
    </Controls-demo.list_new.VirtualScroll.NavigationSwitcher.Index>
</div>
