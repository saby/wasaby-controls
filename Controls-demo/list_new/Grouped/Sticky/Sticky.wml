<div class="controlsDemo__wrapper controlsDemo_fixedWidth350">
<Controls.scroll:Container attr:style="max-height: 200px" shadowMode="js">
  <Controls.list:View
      groupProperty="brand"
      stickyHeader="{{true}}"
      storeId="listData1"/>
</Controls.scroll:Container>
</div>
