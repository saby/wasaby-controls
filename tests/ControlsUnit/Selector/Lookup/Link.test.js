define(['Controls/lookup'], function (lookup) {
   describe('Controls/lookup:Link', function () {
      it('_keyUpHandler', function () {
         var isNotifyClick = false,
            link = new lookup.Link(),
            event = {
               nativeEvent: {
                  keyCode: 13
               }
            };

         link._notify = function (eventName) {
            if (eventName === 'linkClick') {
               isNotifyClick = true;
            }
         };

         link._keyUpHandler(event);
         expect(isNotifyClick).toBe(true);

         isNotifyClick = false;
         link._options.readOnly = true;
         link._keyUpHandler(event);
         expect(isNotifyClick).toBe(false);
      });
   });
});
