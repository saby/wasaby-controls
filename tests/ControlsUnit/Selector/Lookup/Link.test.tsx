/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode } from 'react-dom';
import { Link } from 'Controls/lookup';
import { WasabyEvents } from 'UICore/Events';
import 'ControlsUnit/filterPanelPopup/Sticky/EmptyEditorForTests';
import { render } from '@testing-library/react';

function getLinkBlock(container: HTMLElement): HTMLElement {
    return container.querySelector('.controls-Lookup__link');
}

describe('Controls/lookup:Link', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });
    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        container.remove();
        container = null;
    });

    it('link render', () => {
        render(<Link caption={'caption '} />, { container });
        expect(getLinkBlock(container)).toMatchSnapshot();
    });
});
