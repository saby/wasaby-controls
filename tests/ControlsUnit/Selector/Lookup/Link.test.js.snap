// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls/lookup:Link link render 1`] = `
<span
  class="controls_lookup_theme-undefined controls-Lookup__link
                         controls-Lookup__link-inlineheight-default
                         controls-Lookup__link-default
                         controls-fontsize-lookupLink undefined"
  data-qa="Lookup__link"
  tabindex="-1"
>
  <span
    class="controls-Lookup__link__text
                            controls-Lookup__link_underline-fixed"
  >
    caption 
  </span>
</span>
`;
